lib/finch/http2/pool.ex

```diff
@@ -67,20 +76,76 @@ defmodule Finch.HTTP2.Pool do
 
           :gen_statem.call(pool, {:cancel, ref})
           clean_responses(ref)
-          Process.demonitor(monitor)
+          Process.demonitor(monitor, [:flush])
```

Comment:
This seems correct but I am interested in what lead to you add it here but not above?